src/main/java/com/labsynch/labseer/chemclasses/rdkit/ChemStructureServiceRDKitImpl.java

```diff
@@ -117,67 +127,89 @@ public boolean dropJChemTable(String tableName) {
 	public int[] searchMolStructures(String molfile, String structureTable, String plainTable, String searchType)
 			throws CmpdRegMolFormatException {
 
-			// If preprocessor settings are not null then use the preprocessor settings
-			if(propertiesUtilService.getPreprocessorSettings() != null) {
-	
-				// Read the preprocessor settings as json
-				ObjectMapper objectMapper = new ObjectMapper();
-				JsonNode jsonNode = null;
-				try{
-					jsonNode = objectMapper.readTree(propertiesUtilService.getPreprocessorSettings());
-
-				} catch (IOException e) {
-					logger.error("Error parsing preprocessor settings json: " + propertiesUtilService.getPreprocessorSettings());
-					throw new CmpdRegMolFormatException("Error parsing preprocessor settings json: " + propertiesUtilService.getPreprocessorSettings());
-				}
-	
-				// Extract the url to call
-				JsonNode urlNode = jsonNode.get("processURL");
-				if(urlNode == null || urlNode.isNull()) {
-					logger.error("Missing preprocessorSettings processURL!!");
-				}
-	
-				String url = urlNode.asText();
-	
-				// Get the preprocessory settings
-				ObjectMapper mapper = new ObjectMapper();
-				ObjectNode requestData = mapper.createObjectNode();
-				JsonNode standardizerActions = jsonNode.get("standardizer_actions");
-				ObjectNode options = (ObjectNode) jsonNode.get("process_options");
-				options.put("standardizer_actions", standardizerActions);
-				requestData.put("options", options);
-
-				ArrayNode arrayNode = mapper.createArrayNode();
-				arrayNode.add(molfile);
-				requestData.put("structures", arrayNode);
-		
-				// Post to the service
-				try {
-					String requestString = requestData.toString();
-					logger.info("requestString: " + requestString);
-					String postResponse = SimpleUtil.postRequestToExternalServer(url, requestString, logger);
-					logger.info("Got response: "+ postResponse);
-		
-					// Parse the response json to get the standardized mol
-					ObjectMapper responseMapper = new ObjectMapper();
-					JsonNode responseNode = responseMapper.readTree(postResponse);
-					for (JsonNode responseJsonNode : responseNode) {
-						JsonNode registrationHashesNode = responseJsonNode.get("registration_hash");
-						for (JsonNode registrationHashNode : registrationHashesNode) {
-							logger.info(registrationHashNode.asText());
-						}
-						JsonNode noStereoHashNode = responseJsonNode.get("no_stereo_hash");
-						String noStereoHash = noStereoHashNode.asText();
-						logger.info(noStereoHash);
-					}
+		RDKitStructure serviceRDKitStructure = getRDKitStructureFromService(molfile);
+
+		//Create empty list
+		List<RDKitStructure> rdkitStructures = new ArrayList<RDKitStructure>();
+		if (searchType.equalsIgnoreCase("FULL_TAUTOMER")){
+			rdkitStructures = RDKitStructure.findRDKitStructuresByRegEquals(serviceRDKitStructure.getReg()).getResultList();
```

Comment:
From reading through the CmpdReg FE code and https://github.com/mcneilco/acas-roo-server/blob/release/1.13.6/src/main/java/com/labsynch/labseer/service/RegSearchServiceImpl.java#L76 I think this is the pre-reg search.
This should be doing `findRDKitStructuresByPreRegEquals(serviceRDKitStructure.getPreReg())`

Reply:
Thanks. Makes sense.

Reply:
I fixed this.